lib/active_support/cache/memcached_store.rb

```diff
@@ -128,18 +128,28 @@ def read_multi(*names)
       def cas(name, options = nil)
         options = merged_options(options)
         key = normalize_key(name, options)
+        payload = nil
 
-        handle_exceptions(return_value_on_error: false) do
+        success = handle_exceptions(return_value_on_error: false) do
           instrument(:cas, name, options) do
             @connection.cas(key, expiration(options)) do |raw_value|
               entry = deserialize_entry(raw_value)
               value = yield entry.value
               break true if read_only
-              serialize_entry(Entry.new(value, **options), options)
+              payload = serialize_entry(Entry.new(value, **options), options)
             end
           end
-          true
         end
+
+        success = success != false # can return nil on success
```

Comment:
Couldn't you just leave the `true` handle_exceptions block return value in order to avoid the need for this `success = success != false` line?

Reply:
Ah I see. Good point. Done.